HarClient.Tests/Validation/HarAssert.cs

```diff
@@ -0,0 +1,251 @@
+﻿using HarSharp;
+using Microsoft.VisualBasic;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace JBlam.HarClient.Tests.Validation
+{
+    static class HarAssert
+    {
+        public static void IsValid(Har har) =>
+            IsValidLog(JAssert.HasObjectProperty(har.ToJObject(), "log"));
+        static JObject ToJObject(this object o) =>
+            JObject.FromObject(o, new JsonSerializer
+            {
+                ContractResolver = HarMessageHandler.HarSerializerSettings.ContractResolver,
+                DateFormatHandling = HarMessageHandler.HarSerializerSettings.DateFormatHandling,
+                NullValueHandling = HarMessageHandler.HarSerializerSettings.NullValueHandling,
+            });
+
+        // Regex Copyright (c) 2007 Jan Odvarko
+        // http://www.softwareishard.com/har/viewer/ (schema tab)
+        internal static readonly Regex isoDatetime = new Regex(@"^(\d{4})(-)?(\d\d)(-)?(\d\d)(T)?(\d\d)(:)?(\d\d)(:)?(\d\d)(\.\d+)?(Z|([+-])(\d\d)(:)?(\d\d))");
+
+        static void IsValidLog(JObject log)
+        {
+            _ = JAssert.HasStringProperty(log, "version");
```

Comment:
Consider refactoring this like

```cs
struct JAsserter {
  readonly JObject o;
  public string HasStringProperty(string name) => JAssert.HasStringProperty(o, name);
}
```
etc.?

Reply:
The point being, can we reduce how many times we restate the assertion target's variable name.

Reply:
Committed on a side branch. TBH doesn't look like much of an improvement.